src/ConfigProvider/BundleConfigProvider.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * UrlRewrite Bundle for Contao Open Source CMS.
+ *
+ * @copyright  Copyright (c) 2017, terminal42 gmbh
+ * @author     terminal42 <https://terminal42.ch>
+ * @license    MIT
+ */
+
+namespace Terminal42\UrlRewriteBundle\ConfigProvider;
+
+use Terminal42\UrlRewriteBundle\RewriteConfig;
+
+class BundleConfigProvider implements ConfigProviderInterface
+{
+    /**
+     * @var array
+     */
+    private $entries = [];
+
+    /**
+     * @var string
+     */
+    private $key = 'bundle';
+
+    /**
+     * BundleConfigProvider constructor.
+     *
+     * @param array $entries
+     */
+    public function __construct(array $entries = [])
+    {
+        $this->entries = $entries;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function find(string $id): ?RewriteConfig
+    {
+        list($key, $id) = explode(':', $id);
+
+        // Return if the key is not supported
+        if ($key !== $this->key) {
```

Comment:
Imho that's not correct. Every provider should have a `getKey()` or something. It should not be needed to check here because it's clear that this route is supported.

Reply:
That would work out but the thing is that currently the `RewriteConfig` unique id (based on the provider key) is created inside the provider: https://github.com/terminal42/contao-url-rewrite/pull/3/files#diff-e1a07b28e9d8406d06165e8a252be3b0R104

So if we are about to introduce the `geyKey()` method then the key generation and recognition (supports or not) must be moved out to the rewrite controller and route loader service. Then it's in two places – generation in route loader and the handling in the controller.

Reply:
To me, it should just be like that: I register a new provider and give an array of configs. Whenever the `find()` method is called, it MUST ONLY CONTAIN a routeconfig I provided. It must never be the case that I am called for a routeconfig I did not provide :)

Reply:
Let's check that on Skype tomorrow or the day after 😄